src/plugins/visualizations/public/visualize_app/utils/utils.ts

```diff
@@ -102,5 +102,5 @@ export function getVizEditorOriginatingAppUrl(history: History) {
 }
 
 export function isFallbackDataView(dataView?: DataView): dataView is DataView {
-  return Boolean(dataView && !Object.keys(dataView.getOriginalSavedObjectBody() ?? {}).length);
+  return Boolean(dataView && !dataView.version);
```

Comment:
Should we just use `isPersisted` here? https://github.com/elastic/kibana/blob/560fc6308271056d7776d2815745b0a843e38096/src/plugins/data_views/common/data_views/data_view.ts#L268